### PR TITLE
Feature/mx royale

### PR DESCRIPTION
**Pull request by nihavend**

### CODE REVIEW

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/TextEncoderLiteWrapper.as [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+package mx.utils
+{
+	public class TextEncoderLiteWrapper
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/TextEncoderLiteWrapper.as]: I agree @piotrzarzycki21 , if you confirm 'll make the changes.

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64JSWrapper.as [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+package mx.utils
+{
+	public class Base64JSWrapper
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64JSWrapper.as]: Same for this one @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+    //
+    //--------------------------------------------------------------------------
+
+    /**
+     *  Constant definition for the string "UTF-8".
+     *  
+     *  @langversion 3.0
+     *  @playerversion Flash 9
+     *  @playerversion AIR 1.1
+     *  @productversion Flex 3
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: I'll fix it @piotrzarzycki21.

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+     */
+    public static const CHARSET_UTF_8:String = "UTF-8";
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: I did not change the SWF things  @piotrzarzycki21 . I left them as is.

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+     * denote a new line if <code>insertNewLines</code> is true.
+     * 
+     * The default is <code>10</code> to represent the line feed <code>\n</code>.
+     *  
+     *  @langversion 3.0
+     *  @playerversion Flash 9
+     *  @playerversion AIR 1.1
+     *  @productversion Flex 3
+     */
+    public static var newLine:int = 10;
```
Comment (nihavend) [inline frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: Sure @piotrzarzycki21, it is only related with SWF.

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+    public var insertNewLines:Boolean = true;
+
+    //--------------------------------------------------------------------------
+    //
+    //  Public Methods
+    //
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: okay @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+     */
+    public function drain():String
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: okay @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+        _buffers = [];
+        _buffers.push([]);
+
+        return result;
+    }
+
+    COMPILE::JS
+    public function encode(data:String, encoding = "utf-8"):void
+    {
+        // var bytes:Object = new (TextEncoder || TextEncoderLite)(encoding).encode(data);
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: It was removed from the last edition but i'll check it @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 9]
```diff
@@ -0,0 +1,435 @@
...
+        var base64js:Object = window["base64js"];
+        var textEncoderLite:Object = window["TextEncoderLite"];        
+        
+        var bytes:Object = textEncoderLite(encoding).encode(data);
+        
+        _stringResult = base64js["base64js"].fromByteArray(bytes);
+        */
+
+  		var bytes:Object = new TextEncoderLiteWrapper(encoding).encode(data);
+		_stringResult = new Base64JSWrapper().fromByteArray(bytes);
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: I tried to make minimum change, if you confim it is okay from @piotrzarzycki21

[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: if you confim it is okay from @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+            currentIndex++;
+        }
+
+        data.position = oldPosition;
+    }
+
+    /**
+     * @private
+     */
+    public function flush():String
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: sure @piotrzarzycki21

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as [hunk 11]
```diff
@@ -0,0 +1,435 @@
...
+     * Clears all buffers and resets the encoder to its initial state.
+     *  
+     *  @langversion 3.0
+     *  @playerversion Flash 9
+     *  @playerversion AIR 1.1
+     *  @productversion Flex 3
+     */
+    public function reset():void
+    {
+        _buffers = [];
```
[the remark this answers is not in the document]
Comment (nihavend) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/Base64Encoder.as]: As i said, i trie d to make minimum hcnage in the code and try to  leave as is the SWF side. May i check the this detail?

File: frameworks/projects/MXRoyale/src/main/royale/mx/utils/TextEncoderLiteWrapper.as (hunk 1 above)
[the remark this answers is not in the document]
Comment (piotrzarzycki21) [reply frameworks/projects/MXRoyale/src/main/royale/mx/utils/TextEncoderLiteWrapper.as]: Please make the changes.

**Pull request closed without merging**